src/benchmarks/micro/libraries/System.Net.Security/SslStreamTests.OpenSslAsync.cs

```diff
@@ -19,115 +20,125 @@ namespace System.Net.Security.Tests
         {
-            (NetworkStream client, NetworkStream server) = ConcurrentObjectProvider.CreateIPv4Pair();
-            await DefaultContextHandshake(client, server);
-            client.Dispose();
-            server.Dispose();
+            await Task.Yield();
+            await HandshakeAsync(SslStreamTests._rsa4096Cert, protocol);
         });
 
-        [Benchmark]
-        [BenchmarkCategory(Categories.NoAOT)]
-        public Task ConcurrentDefaultHandshakeContextIPv6Async() => Spawn(ConcurrentTasks, async () =>
+        private static async Task Spawn(int numRequests, int concurrentTasks, Func<Task> method)
         {
-            (NetworkStream client, NetworkStream server) = ConcurrentObjectProvider.CreateIPv6Pair();
-            await DefaultContextHandshake(client, server);
-            client.Dispose();
-            server.Dispose();
-        });
-        
-        [Benchmark]
-        [BenchmarkCategory(Categories.NoAOT)]
-        [ArgumentsSource(nameof(TlsProtocols))]
-        public Task ConcurrentHandshakeContosoAsync(SslProtocols protocol) => Spawn(ConcurrentTasks, async () => await HandshakeAsync(SslStreamTests._cert, protocol));
-
-        [Benchmark]
-        [ArgumentsSource(nameof(TlsProtocols))]
-        [BenchmarkCategory(Categories.NoAOT)]
-        public Task ConcurrentHandshakeECDSA256CertAsync(SslProtocols protocol) => Spawn(ConcurrentTasks, async () => await HandshakeAsync(SslStreamTests._ec256Cert, protocol));
-
-        [Benchmark]
-        [ArgumentsSource(nameof(TlsProtocols))]
-        [BenchmarkCategory(Categories.NoAOT)]
-        [OperatingSystemsFilter(allowed: true, platforms: OS.Linux)]    // Not supported on Windows at the moment.
-        public Task ConcurrentHandshakeECDSA512CertAsync(SslProtocols protocol) => Spawn(ConcurrentTasks, async () => await HandshakeAsync(SslStreamTests._ec512Cert, protocol));
+            var _tasks = new Collections.Generic.List<Func<Task>>(concurrentTasks);
 
-        [Benchmark]
-        [ArgumentsSource(nameof(TlsProtocols))]
-        [BenchmarkCategory(Categories.NoAOT)]
-        public Task ConcurrentHandshakeRSA2048CertAsync(SslProtocols protocol) => Spawn(ConcurrentTasks, async () => await HandshakeAsync(SslStreamTests._rsa2048Cert, protocol));
-
-        [Benchmark]
-        [ArgumentsSource(nameof(TlsProtocols))]
-        public Task ConcurrentHandshakeRSA4096CertAsync(SslProtocols protocol) => Spawn(ConcurrentTasks, async () => await HandshakeAsync(SslStreamTests._rsa4096Cert, protocol));
+            for(int j = 0; j < numRequests; ++j)
+            {
+                for(int i = 0; i < concurrentTasks; ++i)
+                {
+                    _tasks.Add(method);
+                }
 
-        private static async Task Spawn(int count, Func<Task> method)
-        {
-            var _tasks = new Collections.Generic.List<Task>(count);
+                await Task.WhenAll(_tasks.AsParallel().Select(async task => await task()));
```

Comment:
@anthonycanino @clairiky

`await Task.Yield(); // forces task to become async` yet it doesn't guarantee the task to run in parallel in different thread. [StackOverflow](https://stackoverflow.com/questions/34375696/executing-tasks-in-parallel#:~:text=If%20you%20want,other%20tasks%0A%7D)
So, besides adding `Task.Yield()` I've added this line too.